File: NettyServer/src/com/netty/test/event/TestEvent.java
package com.netty.test.event;

import org.msgpack.annotation.Index;
import org.msgpack.annotation.Message;

import com.netty.test.util.TimeUtil;

/**
 * 测试事件
 * 
 * @author lucher
 */
@Message // 使用MessagePack传输的实体类需要加入该注解
public class TestEvent extends BaseEvent {

	// 需要有默认构造，否则会出现错误：Caused by: compile error: no such constructor:
	// com.netty.test.event.TestEvent
	public TestEvent() {
	}

	public TestEvent(EventType eventType, String from, String content) {
		setEventType(eventType);
		setFrom(from);
		setContent(content);
		setTime(TimeUtil.getCurrentTime("yyyy-MM-dd HH:mm:ss"));
	}

	// 消息内容
	@Index(3)
	private String content;

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	@Override
	public String toString() {
		return String.format("【%s,%s】", time, content);
	}
}

File: NettyCSharp/com.netty.csharp/event/TestEvent.cs
#region << 版 本 注 释 >>
/*----------------------------------------------------------------
* 类 名 称 ：TestEvent
* 类 描 述 ：测试事件
* 作    者 ：lucher
* 版 本 号 ：v1.0.0
*******************************************************************
* Copyright @ lucher 2019. All rights reserved.
*******************************************************************
//----------------------------------------------------------------*/
#endregion
using MessagePack;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NettyCSharp
{
    [MessagePackObject]// 使用MessagePack传输的实体类需要加入该注解
    public class TestEvent:BaseEvent
    {
        public TestEvent() { }

        public TestEvent(EventType eventType, string from, string content)
        {
            this.SetEventType(eventType);
            this.SetFrom(from);
            this.SetContent(content);
            this.SetTime(DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss")); ;
        }

        // 消息内容
        [Key(3)]
        public String content;

        public string GetContent()
        {
            return content;
        }

        public void SetContent(string value)
        {
            content = value;
        }

        public override string ToString()
        {
            return String.Format("【{0},{1}】", time, content);
        }
    }
}
